api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -198,17 +198,20 @@ message HostCertsRequest {
 // OpenSSHCertRequest specifies certificate-generation parameters
 // for a certificates used to connect to Agentless nodes.
 message OpenSSHCertRequest {
-  // Username is the Teleport username.
-  string Username = 1 [(gogoproto.jsontag) = "username"];
+  // Username is the Teleport user the certificate will be generated for.
+  types.UserV2 User = 1 [(gogoproto.jsontag) = "user"];
+  // Roles are the roles of the Teleport user the certificate will be
+  // generated for.
+  repeated types.RoleV6 Roles = 2 [(gogoproto.jsontag) = "roles"];
   // PublicKey is the public key to sign.
-  bytes PublicKey = 2 [(gogoproto.jsontag) = "public_key"];
+  bytes PublicKey = 3 [(gogoproto.jsontag) = "public_key"];
   // TTL is the duration the certificate will be valid for.
-  int64 TTL = 3 [
+  int64 TTL = 4 [
     (gogoproto.jsontag) = "ttl",
     (gogoproto.casttype) = "Duration"
   ];
   // Cluster is the Teleport cluster name.
-  string Cluster = 4 [(gogoproto.jsontag) = "cluster"];
+  string Cluster = 5 [(gogoproto.jsontag) = "cluster"];
```

Comment:
Has there been a release with this protobuf message yet? If so, we can't change the numbers here.

Reply:
I'm not sure what the right thing to do in this situation is since it looks like it was broken already, but maybe we should just shift all of the field numbers up and leave the old ones. Start with user at 5, for example. @espadolini would probably know for sure.

Reply:
Yeah, just remove and reserve `1` and `Username` and add the new fields as 5 and 6.

Any field number under 16 encodes as a single byte, anyway.

Reply:
Ohh, I figured since agentless hasn't been released it this wouldn't be a breaking change but didn't consider the fact that the message is still in `master`... I'll fix that

Reply:
Done